_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp

```diff
@@ -4398,18 +4398,21 @@ mfxStatus MfxHwH264Encode::CheckVideoParamQueryLike(
         changed = true;
     }
 
+    bool mfxRateControlCbrOnBxt = 
+        (platform == MFX_HW_APL || platform == MFX_HW_CFL) && 
+        (par.mfx.RateControlMethod == MFX_RATECONTROL_CBR);
+
+    bool mfxRateControl =
+            par.mfx.RateControlMethod == MFX_RATECONTROL_LA  ||
+            par.mfx.RateControlMethod == MFX_RATECONTROL_LA_HRD ||
+            par.mfx.RateControlMethod == MFX_RATECONTROL_LA_EXT ||
+            par.mfx.RateControlMethod == MFX_RATECONTROL_VBR ||
+            par.mfx.RateControlMethod == MFX_RATECONTROL_QVBR ||
+            (mfxRateControlCbrOnBxt && !IsOn(extOpt2->ExtBRC));
+
      if (extOpt3->WinBRCMaxAvgKbps || extOpt3->WinBRCSize)
      {
-         if ((par.mfx.RateControlMethod != MFX_RATECONTROL_LA  &&
```

Comment:
@stasefremov : Stas, do you understand what is written here?! or do you know someone who does? I very much want to see list of modes for which we try to apply settings rather than list of modes for which we don't! I simply can't say what we should do without knowing the intent...

Reply:
@takulako , perhaps you can help. Can you please clarify ?

Reply:
The old check looks clear enough for me:
Sliding window supported by limited number of RC modes (LA, LA_HRD, LA_EXT, VBR, QVBR, HW CBR).
If the current mode is not one of them - zero all winbrc parameters and return warning.

What's unclear here is the proposed changes - mfxRateControl variable name looks confusing and doesn't reflect its actual sense.

Reply:
mfxRateControl  renamed to slidingWindowSupported

Reply:
@eistarov : for me the code is not clear. It becomes clear after your comment which is entirely missing in the code. I suggest to add exactly what you have wrote: "Sliding window is supported by limited number of RC modes (LA, LA_HRD, LA_EXT, VBR, QVBR, HW CBR)." After that code starts to make some sense. @IlyaSergeyev : please, add.